pkg/ccl/changefeedccl/sink.go

```diff
@@ -268,15 +318,15 @@ func (s errorWrapperSink) EmitRow(
 func (s errorWrapperSink) EmitResolvedTimestamp(
 	ctx context.Context, encoder Encoder, resolved hlc.Timestamp,
 ) error {
-	if err := s.wrapped.EmitResolvedTimestamp(ctx, encoder, resolved); err != nil {
+	if err := s.wrapped.(ResolvedTimestampSink).EmitResolvedTimestamp(ctx, encoder, resolved); err != nil {
```

Comment:
perhaps avoid this casting and just have errorWrapperEventSink/errorWrapperResolvedSink....

you might be able to remove "Sink" interface.

Reply:
Will do in a followup.